opentelemetry/src/logs/logger.rs

```diff
@@ -30,128 +30,37 @@ pub trait LoggerProvider {
-        }
-        if let Some(s) = schema_url {
-            builder = builder.with_schema_url(s);
-        }
-        if let Some(a) = attributes {
-            builder = builder.with_attributes(a);
-        }
-        builder.build()
-    }
-
-    /// Returns a new builder for creating a [`Logger`] instance
-    ///
-    /// The `name` should be the application name or the name of the library
-    /// providing instrumentation. If the name is empty, then an
-    /// implementation-defined default name may be used instead.
+    /// Returns a new logger with the given instrumentation library.
     ///
     /// # Examples
     ///
     /// ```
-    /// use opentelemetry::InstrumentationLibrary;
-    /// use crate::opentelemetry::logs::LoggerProvider;
+    /// use opentelemetry::InstrumentationScope;
+    /// use opentelemetry::logs::LoggerProvider;
     /// use opentelemetry_sdk::logs::LoggerProvider as SdkLoggerProvider;
     ///
     /// let provider = SdkLoggerProvider::builder().build();
     ///
     /// // logger used in applications/binaries
-    /// let logger = provider.logger_builder("my_app").build();
+    /// let logger = provider.logger("my_app");
     ///
     /// // logger used in libraries/crates that optionally includes version and schema url
-    /// let logger = provider.logger_builder("my_library")
+    /// let scope = InstrumentationScope::builder(env!("CARGO_PKG_NAME"))
     ///     .with_version(env!("CARGO_PKG_VERSION"))
     ///     .with_schema_url("https://opentelemetry.io/schema/1.0.0")
     ///     .build();
-    /// ```
-    fn logger_builder(&self, name: impl Into<Cow<'static, str>>) -> LoggerBuilder<'_, Self> {
-        LoggerBuilder {
-            provider: self,
-            library_builder: InstrumentationLibrary::builder(name),
-        }
-    }
-
-    /// Returns a new versioned logger with the given instrumentation library.
-    ///
-    /// # Examples
     ///
+    /// let logger = provider.library_logger(scope);
     /// ```
-    /// use opentelemetry::InstrumentationLibrary;
-    /// use crate::opentelemetry::logs::LoggerProvider;
-    /// use opentelemetry_sdk::logs::LoggerProvider as SdkLoggerProvider;
-    ///
-    /// let provider = SdkLoggerProvider::builder().build();
-    ///
-    /// // logger used in applications/binaries
-    /// let logger = provider.logger("my_app");
-    ///
-    /// // logger used in libraries/crates that optionally includes version and schema url
-    /// let library = std::sync::Arc::new(
-    ///     InstrumentationLibrary::builder(env!("CARGO_PKG_NAME"))
-    ///         .with_version(env!("CARGO_PKG_VERSION"))
-    ///         .with_schema_url("https://opentelemetry.io/schema/1.0.0")
-    ///         .build(),
-    /// );
-    /// let logger = provider.library_logger(library);
-    /// ```
-    fn library_logger(&self, library: Arc<InstrumentationLibrary>) -> Self::Logger;
+    fn library_logger(&self, scope: InstrumentationScope) -> Self::Logger;
```

Comment:
we need to rename `library_logger` to something more indicative of its purpose...
provider.logger("name")
provider.logger_with_scope(scope);

open to suggestions.

Reply:
`logger_with_scope` looks fine to me.